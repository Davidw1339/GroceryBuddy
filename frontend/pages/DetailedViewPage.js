import React, {Component} from 'react';
import {searchByUPC} from "../utils/api";
import {View, ActivityIndicator, StyleSheet, FlatList} from 'react-native';
import _ from 'lodash'
import { Image, PricingCard, Overlay, Text, Input, Button } from 'react-native-elements';

export default class DetailedViewPage extends Component {
    constructor(props) {
        super(props);
        store = ''
        //Had to wrap this in a try catch for when the component is loaded with no navigation (testing)
        try {
            store = this.props.navigation.state.params.store
        } catch(err) {
            console.log(err)
        }
        this.state = {
            isLoading:true,
            priceOverlayVisible: false,
            newPrice: '',
            store: store
        }
    }

    async componentDidMount() {
        let data = await searchByUPC(this.props.navigation.state.params.upc);

        //we only want the prices for the store we selected
        currStore = this.state.store
        elem = data[0]
        stores = elem.stores
        res = _.find(stores, ['name', currStore])
        oldPrices = res.prices

        //add oldprices, as well as other item data, and let the app load
        this.setState({isLoading: false, oldPrices: oldPrices, ...data[0]})
    }

    toggleChangePriceOverlay = () => {
        this.setState(({priceOverlayVisible}) => ({
            priceOverlayVisible: !priceOverlayVisible
        }));
    }


    //TODO
    changePrice = () => {

    }

    render() {
        if (this.state.isLoading) {
            return (
                <View style={{flex: 1, alignContent: 'center', justifyContent: 'center'}}>
                    <ActivityIndicator/>
                </View>
            );
        }

        return (
          <View style={styles.DetailContainer}>
            <View style={styles.ItemImage}>
                <Image
                    PlaceholderContent={<ActivityIndicator />}
                    resizeMode='stretch'
                    style={{flex:1, width: 150, height: 150}}
                    source={{
                        uri:
                            this.state.image_url,
                    }} />
            </View>
            <View style={styles.HistoryContainer}>
                <View style={styles.HistoryHeader}>
                        <Text style={{ fontSize: 20, marginBottom: 0, paddingBottom: 0}}>History of prices at </Text>
                        <Text style={{ fontSize: 20, fontWeight: 'bold', marginBottom: 0, paddingBottom: 0}}>{this.state.store}:</Text>
                </View>
                <View style={{flex: 1}}>
                    <FlatList
                        data={this.state.oldPrices}
                        renderItem={({item}) => 
                            <View style={styles.OldPriceListItem}>
                                <Text style={{alignSelf: "flex-start"}}>{item.price.toString()}</Text>
                                <Text style={{ alignSelf: "flex-end", color: 'grey' }}>
                                    {((new Date(Math.floor(item.date) * 1000)).getMonth() + 1) + '/' + 
                                      (new Date(Math.floor(item.date) * 1000)).getDate() + '/' + 
                                      (new Date(Math.floor(item.date) * 1000)).getFullYear()}
                                </Text>
                            </View>}
                        keyExtractor={(item, index) => 'key' + index}
                    />
                </View>
            </View>
            <View style={styles.priceContainer}>
                <PricingCard
                    color="#4f9deb"
                    title={this.state.name}
                    price={"$" + this.state.oldPrices[0].price.toString()}
                    button={{ title: 'Add/Change Price', icon: 'add' }}
                    onButtonPress={this.toggleChangePriceOverlay}
                />
            </View>
            <Overlay
                isVisible={this.state.priceOverlayVisible}
                onBackdropPress={this.toggleAddListOverlay}
                animationType="fade"
                height={150}>
                <View style={styles.simpleContainer}>
                    <View style={styles.simpleContainer}>
                        <Text style={styles.button}>New Price:</Text>
                        <Input
                            value={this.state.newPrice}
                            onChangeText={(newPrice) => this.setState({newPrice})}
                            placeholder="$$$$"/>
                    </View>
                    <View style={styles.buttonGroup}>
                        <Button containerStyle={styles.dialogButton} title="CANCEL" onPress={this.toggleChangePriceOverlay} />
                        <Button containerStyle={styles.dialogButton} title="SUBMIT" onPress={this.toggleChangePriceOverlay} />
                    </View>
                </View>
            </Overlay>
          </View>
        );
    }
}

const styles = StyleSheet.create({
    DetailContainer: {
        flex:1,
        flexDirection: 'column',
        justifyContent: 'center', 
        alignItems: 'center'
    },
    HistoryContainer: {
        flex: 2,
        width: '100%',
    },
    HistoryHeader: {
        flexDirection: 'row', 
        paddingLeft: 5, 
        paddingRight: 5, 
        marginBottom: 0, 
        paddingBottom: 0, 
        height: 30
    },
    OldPriceListItem: {
        flexDirection: 'row', 
        justifyContent: 'space-between', 
        paddingLeft: 10, 
        paddingRight: 10, 
        paddingBottom: 5
    },
    ItemImage: {
        flex:3,
        paddingTop: 5
    },
    priceContainer: {
        flex: 0,
    },
    simpleContainer: {
        flex: 1
    },
    dialogButton: {
        flex: 1,
        margin: 5,
        borderRadius: 100
    },
    button: {
        margin: 5
    },
    buttonGroup: {
        flexDirection: "row"
    },
});