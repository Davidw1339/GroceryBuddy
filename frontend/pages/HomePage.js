import React from 'react';
import { View, StyleSheet, ScrollView } from 'react-native';
import {
  Text, Button, Overlay, Input,
} from 'react-native-elements';
import ActionButton from 'react-native-action-button';
import ListItem from '../components/ListItem';
import { getLists, addList, deleteList } from '../utils/api';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f4f4f4',
    padding: 10,
  },
  button: {
    margin: 5,
  },
  scrollView: {
    flex: 1,
  },
  simpleContainer: {
    flex: 1,
  },
  buttonGroup: {
    flexDirection: 'row',
  },
  dialogButton: {
    flex: 1,
    margin: 5,
    borderRadius: 100,
  },
});

export default class HomePage extends React.Component {
    static navigationOptions = {
      title: 'Lists',
    }

    constructor(props) {
      super(props);
      this.state = {
        lists: [],
        listOverlayVisible: false,
        newListName: '',
      };
    }

    componentDidMount() {
      this.retrieveLists();
    }

    retrieveLists = async () => {
      const lists = await getLists();
      this.setState({
        lists,
      });
    }

    toggleAddListOverlay = () => { // toggles the overlay/modal for creating new list
      this.setState(({ listOverlayVisible }) => ({
        listOverlayVisible: !listOverlayVisible,
      }));
    }

    addNewList = async () => {
      this.toggleAddListOverlay(); // close the modal
      this.setState({ newListName: '' }); // clear our the modal input
      await addList(this.state.lists, this.state.newListName); // add the new list
      this.retrieveLists(); // refresh the set of lists on the homepage
    }

    deleteList = async (id) => {
      await deleteList(this.state.lists, id);
      this.retrieveLists();
    }

    launchList = (name) => {
      this.props.navigation.navigate('Search', { name });
    }

    render() {
      return (
        <View style={styles.container}>
          {/* Render all list items */}
          <ScrollView style={styles.scrollView}>
            {this.state.lists.map(({ name }, i) => (
              <ListItem
                key={name + i}
                name={name}
                handleDelete={() => { this.deleteList(i); }}
                handleLaunchList={() => { this.launchList(name); }}
              />
            ))}
          </ScrollView>
          <ActionButton buttonColor="#2196f3" onPress={this.toggleAddListOverlay} />

          {/* This overlay is shown to read in user input and create a new list */}
          <Overlay
            isVisible={this.state.listOverlayVisible}
            onBackdropPress={this.toggleAddListOverlay}
            animationType="fade"
            height={150}
          >
            <View style={styles.simpleContainer}>
              <View style={styles.simpleContainer}>
                <Text style={styles.button}>Name of List:</Text>
                <Input
                  value={this.state.newListName}
                  onChangeText={newListName => this.setState({ newListName })}
                  placeholder="List Name"
                />
              </View>
              <View style={styles.buttonGroup}>
                <Button containerStyle={styles.dialogButton} title="CANCEL" onPress={this.toggleAddListOverlay} />
                <Button containerStyle={styles.dialogButton} title="ADD" onPress={this.addNewList} />
              </View>
            </View>
          </Overlay>
        </View>
      );
    }
}
