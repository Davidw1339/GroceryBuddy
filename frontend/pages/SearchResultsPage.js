import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Text } from 'react-native-elements';

export default class SearchResultsPage extends React.Component {
    static navigationOptions = {
        title: 'Search Results Page',
    };

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <View style={styles.container}>
                <Text>Search Results Page</Text>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        paddingTop: 30,
        padding: 10
    }
});