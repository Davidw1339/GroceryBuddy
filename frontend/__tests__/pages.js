import React from 'react';
import renderer from "react-test-renderer";
import DetailedViewPage from "../pages/DetailedViewPage";
import HomePage from "../pages/HomePage";
import SearchPage from "../pages/SearchPage";
import SearchResultsPage from "../pages/SearchResultsPage";
import ShoppingPage from "../pages/ShoppingPage";



test('DetailViewPage', () => {
    jest.useFakeTimers();
    const tree = renderer.create(<DetailedViewPage />).toJSON();
    expect(tree).toMatchSnapshot();
});


test('HomePage', () => {
    jest.useFakeTimers();
    const tree = renderer.create(<HomePage />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('SearchPage', () => {
    jest.useFakeTimers();
    const tree = renderer.create(<SearchPage />).toJSON();
    expect(tree).toMatchSnapshot();
});


