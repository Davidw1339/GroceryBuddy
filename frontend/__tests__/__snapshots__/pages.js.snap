// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailViewPage: DetailViewPage 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;

exports[`HomePage: HomePage 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f4f4f4",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "elevation": undefined,
            "justifyContent": "flex-end",
            "zIndex": undefined,
          },
        ],
        undefined,
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "elevation": undefined,
          "justifyContent": "flex-end",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": undefined,
        }
      }
    />
    <View
      pointerEvents="box-none"
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "elevation": undefined,
              "justifyContent": "flex-end",
              "zIndex": undefined,
            },
          ],
          Object {
            "alignItems": "flex-end",
          },
          Object {
            "paddingVertical": 30,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginHorizontal": 30,
              "zIndex": undefined,
            },
            Object {
              "elevation": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.35,
              "shadowRadius": 3,
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(33, 150, 243, 1)",
                "borderRadius": 28,
                "height": 56,
                "width": 56,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 28,
                  "height": 56,
                  "justifyContent": "center",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                  "width": 56,
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(255, 255, 255, 1)",
                    "fontSize": 24,
                    "marginTop": -4,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, .4)",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
      testID="RNE__Overlay__backdrop"
    />
    <View
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 3,
            "height": 150,
            "padding": 10,
            "shadowColor": "rgba(0, 0, 0, .3)",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowRadius": 4,
            "width": 670,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "margin": 5,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Name of List:
            </Text>
            <View
              style={
                Object {
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "#86939e",
                    "flexDirection": "row",
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  placeholder="List Name"
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "black",
                      "flex": 1,
                      "fontSize": 18,
                      "minHeight": 40,
                    }
                  }
                  testID="RNE__Input__text-input"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 100,
                  "flex": 1,
                  "margin": 5,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#2089dc",
                      "borderColor": "#2089dc",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingBottom": 1,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    CANCEL
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderRadius": 100,
                  "flex": 1,
                  "margin": 5,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#2089dc",
                      "borderColor": "#2089dc",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingBottom": 1,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    ADD
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`SearchPage: SearchPage 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "lightgray",
        "borderBottomWidth": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 1,
          "paddingBottom": 8,
          "paddingTop": 8,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 0,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 8,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text />
              </View>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            onBlur={[Function]}
            onCancel={[Function]}
            onChangeText={[Function]}
            onClear={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Grocery Item Name Here"
            platform="android"
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "marginLeft": 24,
                "marginRight": 8,
                "minHeight": 40,
              }
            }
            testID="searchInput"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 15,
                "marginRight": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 3,
          "height": 35,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "#bcbbc1",
            "borderTopWidth": 0.5,
            "flexDirection": "row",
            "padding": 14,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "fontSize": 17,
              }
            }
            testID="listItemTitle"
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            Total (0) items
          </Text>
        </View>
        <View
          width={16}
        />
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 0.5,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 17,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            $0.00
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Finish Shopping List
          </Text>
          <View
            style={
              Object {
                "marginHorizontal": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
              underlayColor="white"
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
